test-frame-metrics-collector/src/main/java/io/skodjob/testframe/MetricsCollector.java

```diff
@@ -0,0 +1,565 @@
+     * of the specified pattern. If the metric is not initially available, this method periodically retries
+     * the collection until the metric appears or a timeout occurs.
+     *
+     * @param pattern       The regular expression pattern used to identify and extract the metric.
+     * @return              A list of collected metric values. If no metrics are found, the returned list will be empty.
+     */
+    public final synchronized ArrayList<Double> waitForSpecificMetricAndCollect(Pattern pattern) {
+        ArrayList<Double> values = collectSpecificMetric(pattern);
+
+        if (values.isEmpty()) {
+            Wait.until(String.format("metrics contain pattern: %s", pattern.toString()),
+                TestFrameConstants.GLOBAL_POLL_INTERVAL_MEDIUM, TestFrameConstants.GLOBAL_TIMEOUT, () -> {
+                try {
+                    this.collectMetricsFromPods();
+                } catch (MetricsCollectionException e) {
+                    throw new RuntimeException(e);
+                }
+                LOGGER.debug("Collected data: {}", this.collectedData);
+                ArrayList<Double> vals = this.collectSpecificMetric(pattern);
+
+                if (!vals.isEmpty()) {
+                    values.addAll(vals);
+                    return true;
+                }
+
+                return false;
+            });
+        }
+
+        return values;
+    }
+
+    /**
+     * Executes a command to collect metrics from a specific pod.
+     * The method constructs and executes a shell command to scrape metrics using curl,
+     * handling the execution within a specified timeout.
+     *
+     * @param metricsPodIp              The IP address of the metrics pod.
+     * @param podName                   The name of the pod from which metrics are being collected.
+     * @return String                   The output from the executed command, typically metrics data.
+     * @throws InterruptedException     If the thread is interrupted during command execution.
+     * @throws ExecutionException       If an error occurs during the command execution.
+     * @throws IOException              If an I/O error occurs during command handling.
+     */
+    protected String collectMetrics(String metricsPodIp, String podName)
+        throws InterruptedException, ExecutionException, IOException {
+        List<String> executableCommand = Arrays.asList(getKubeCmdClient().inNamespace(namespaceName).toString(), "exec",
+            scraperPodName,
+            "-n", namespaceName,
+            "--", "curl", metricsPodIp + ":" + component.getDefaultMetricsPort() + component.getDefaultMetricsPath());
+
+        LOGGER.debug("Executing command:{} for scrape the metrics", executableCommand);
+
+        // 20 seconds should be enough for collect data from the pod
+        int ret = this.exec.execute(null, executableCommand, null,20_000);
```

Comment:
```suggestion
        int ret = this.exec.execute(null, executableCommand, null, 20_000);
```

Wouldn't be better to use some constant as a timeout?

Reply:
Yeah true updated :)